src/main/java/org/mockito/DoNotMock.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright (c) 2019 Mockito contributors
+ * This program is made available under the terms of the MIT License.
+ */
+package org.mockito;
+
+import static java.lang.annotation.ElementType.TYPE;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+
+/**
+ * Annotation representing a type that should not be mocked.
+ * <p>When marking a type {@code @DoNotMock}, you should always point to alternative testing
+ * solutions such as standard fakes or other testing utilities.
+ *
+ * Mockito enforces {@code @DoNotMock} with the {@link org.mockito.plugins.DoNotMockEnforcer}.
+ *
+ * If you want to use a custom {@code @DoNotMock} annotation, the {@link org.mockito.plugins.DoNotMockEnforcer}
+ * will match on annotations with a type ending in "org.mockito.DoNotMock". You can thus place
+ * your custom annotation in {@code com.my.package.org.mockito.DoNotMock} and Mockito will enforce
+ * that types annotated by {@code @com.my.package.org.mockito.DoNotMock} can not be mocked.
+ */
```

Comment:
It would be nice to add a simple code snippet with the use of the reason. 

Since annotation are likely to be how user discover this feature, it may makes sense to me to also describe how to make non-mockable the non-annotatable types like `File`.

Reply:
I have added an example, good suggestion 👍 

I think we can ship this without adding classes like `File` in Mockito. Then once we know the enforcer is working as expected, we can make the switch in the next major version of Mockito to by default disallow mocking these classes.